### DISCUSSION TITLE
(shinyapps) Saat men running di local bisa tetapi saat di publish plot tidak muncul

### DISCUSSION THREAD

**Discussion by RefsyR** [teamalgoritma/community#725] [Q&A] [answered]
![image](https://github.com/teamalgoritma/community/assets/130164698/e75ff6d6-725d-4a93-83ca-84a9991a5da2)

Saat men running di local bisa tetapi saat di publish plot tidak muncul

**Comment by ramdhinafinita** [comment]
Dear Pak @RefsyR 
Bisa dilakukan publish ulang dengan membuat Rproject baru, data UI.R dan lainnya bisa dimasukkan pada Rproj baru tersebut dan coba publish ulang dengan nama yang berbeda.  Karena, error pada  logs nya menyatakan object yang digunakan untuk pembuatan dashboard tidak terpanggil.

**Comment by RefsyR** [comment]
Masih tetap tidak bisa kak

**Reply by ramdhinafinita** [reply]
Bisa coba kirimkan full zip nya pak? agar kami bisa cek?

**Comment by RefsyR** [comment]
[Uploading Capstone1.zip…]()

**Reply by ramdhinafinita** [reply]
Maaf pak, untuk file nya tidak bisa di download.  Bisa coba diulang?

**Reply by RefsyR** [reply]
tadi saya kirim melalui email

**Reply by ramdhinafinita** [reply]
Email siapa ya pak? di email [mentor@algorit.ma] masih belum ada kiriman file.  Mungkin bisa di cek kembali

**Reply by RefsyR** [reply]
sudah saya kirim ke mentor@algorit.ma email tersebut

**Comment by RefsyR** [comment]
[Capstone1.zip](https://github.com/teamalgoritma/community/files/12068744/Capstone1.zip)
,

**Comment by ramdhinafinita** [comment, marked as answer]
Dear Pak @RefsyR 
Kami sudah mencoba untuk mengakses folder yang Bapak lampirkan dan melakukan running local, tetapi hasil yang dikeluarkan sama persis seperti hasil deploy shiny. Berikut error yang dihasilkan dan saran perbaikan: 
![image (2)](https://github.com/teamalgoritma/community/assets/64061969/7bfae237-c0e3-4de0-a18d-6a8c3a005d3c)

1. Error dikarenakan Bapak memanggil data pada file single app /app.R yang seharusnya jika menggunakan single app, bapak tidak perlu membuat file server dan ui secara terpisah. Jika ingin dipisah, perlu membuat global.R bukan app.R

![image (3)](https://github.com/teamalgoritma/community/assets/64061969/b030fa9a-460e-4f2d-883e-cbf86a7c0e86)


2. Read data tidak perlu menggunakan test1/ karena file .R sudah sejajar dengan file csv.

3. Cek kembali untuk plot  3 yang menampilkan kasus positif, pada server tidak ada pemanggilan object confirmed. Ini dikarenaka plot_ly yang bapak gunakan tidak memanggil object nya hanya ada read.csv pada app.R yang seharusnya dimasukkan kedalam server.R sebelum:

server<-shinyServer(function(input, output, session){

Komentar selanjutnya bisa dicek pada google classroom capstone Bapak. Dimana bapak perlu melakukan revisi sesuai dengan instruksi yang sudah kami berikan.
